File: java/src/test/java/fr/soat/cleancoders/exo1/MatcherTest.java
package fr.soat.cleancoders.exo1;

import org.assertj.core.api.Assertions;
import org.junit.Before;
import org.junit.Test;

public class MatcherTest {
    private final int clipLimit = 100;
    private final int delta = 5;
    private Matcher matcher;
    private int[] expected;

    @Before
    public void initialize_common_variables() {
        matcher = new Matcher();
        expected = new int[]{10, 50, 30, 98};
    }

    @Test
    public void should_match_when_the_difference_between_the_actual_entry_and_the_expected_one_is_below_delta() {
        //Given
        int[] actual = new int[]{12, 55, 25, 110};

        //When
        boolean matches = matcher.match(expected, actual, clipLimit, delta);

        //Then
        Assertions.assertThat(matches).isTrue();

    }

    @Test
    public void should_not_match_when_the_difference_between_the_actual_entry_and_the_expected_one_is_above_delta() {
        //Given
        int[] actual = new int[]{10, 60, 30, 98};

        //When
        boolean matches = matcher.match(expected, actual, clipLimit, delta);

        //Then
        Assertions.assertThat(matches).isFalse();

    }

    @Test
    public void should_not_match_when_the_actual_and_the_expected_have_different_length() {
        //Given
        int[] actual = new int[]{10, 50, 30};

        //When
        boolean matches = matcher.match(expected, actual, clipLimit, delta);

        //Then
        Assertions.assertThat(matches).isFalse();

    }
}

File: csharp/Exo1/MatcherTest.cs
using NFluent;
using Xunit;

namespace Soat.CleanCoders.Refactoring.Exo1
{
    public class MatcherTest
    {
        private const int ClipLimit = 100;
        private const int Delta     = 5;

        private readonly Matcher _sut = new Matcher();

        [Fact]
        public void Should_match_given_no_difference_and_values_below_clip_limit()
        {
            //Given
            var actual   = new[] { 10, 50, 30, 90 };
            var expected = new[] { 10, 50, 30, 90 };

            //When
            var matches = _sut.Match(expected, actual, ClipLimit, 0);

            //Then
            Check.That(matches)
                 .IsTrue();
        }

        [Fact]
        public void Should_match_given_difference_below_delta()
        {
            //Given
            var actual   = new[] { 12, 55, 25, 110 };
            var expected = new[] { 10, 50, 30, 98 };

            //When
            var matches = _sut.Match(expected, actual, ClipLimit, Delta);

            //Then
            Check.That(matches)
                 .IsTrue();
        }

        [Fact]
        public void Should_not_match_given_difference_above_delta()
        {
            //Given
            var actual   = new[] { 10, 60, 30, 98 };
            var expected = new[] { 10, 50, 30, 98 };

            //When
            var matches = _sut.Match(expected, actual, ClipLimit, Delta);

            //Then
            Check.That(matches)
                 .IsFalse();
        }

        [Fact]
        public void Should_not_match_given_different_length()
        {
            //Given
            var actual   = new[] { 10, 50, 30 };
            var expected = new[] { 10, 50, 30, 98 };

            //When
            var matches = _sut.Match(expected, actual, ClipLimit, Delta);

            //Then
            Check.That(matches)
                 .IsFalse();
        }
    }
}
